src/Microsoft.Health.Dicom.Core/Features/Indexing/InstanceReindexer.cs

```diff
@@ -23,19 +23,29 @@ public class InstanceReindexer : IInstanceReindexer
     {
         private readonly IMetadataStore _metadataStore;
         private readonly IIndexDataStore _indexDataStore;
+        private readonly IDicomDatasetReindexValidator _dicomDatasetReindexValidator;
 
-        public InstanceReindexer(IMetadataStore metadataStore, IIndexDataStore indexDataStore)
+        public InstanceReindexer(IMetadataStore metadataStore, IIndexDataStore indexDataStore, IDicomDatasetReindexValidator dicomDatasetReindexValidator)
         {
             _metadataStore = EnsureArg.IsNotNull(metadataStore, nameof(metadataStore));
             _indexDataStore = EnsureArg.IsNotNull(indexDataStore, nameof(indexDataStore));
+            _dicomDatasetReindexValidator = EnsureArg.IsNotNull(dicomDatasetReindexValidator, nameof(dicomDatasetReindexValidator));
         }
 
         public async Task ReindexInstanceAsync(IReadOnlyCollection<ExtendedQueryTagStoreEntry> entries, VersionedInstanceIdentifier versionedInstanceId, CancellationToken cancellationToken)
         {
             EnsureArg.IsNotNull(entries, nameof(entries));
             EnsureArg.IsNotNull(versionedInstanceId, nameof(versionedInstanceId));
+            if (entries.Count == 0)
```

Comment:
When would we be in ReIndexing and not have any tags to re-index?

Reply:
The intention is to avoid Blob read when there is no query tags (query tag could be deleted during reindex).
But I checked the logic, it's impossible `entries.Count ==0`
will update.